test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests_CSharp.cs

```diff
@@ -231,14 +231,83 @@ public async Task ArmExtensionsResourceFilter_GetNonSecretResource_Succeeds()
+        [InlineData("admin/host/status", true, true, true, false, false, HttpStatusCode.Unauthorized)]
+        [InlineData("admin/host/status", true, true, true, true, true, HttpStatusCode.OK)]
+        [InlineData("admin/host/status", true, false, false, false, true, HttpStatusCode.OK)]
+        [InlineData("admin/host/status", true, false, false, true, true, HttpStatusCode.OK)]
+        [InlineData("admin/host/status", true, true, true, false, true, HttpStatusCode.OK)]
+        [InlineData("admin/host/status", false, true, false, true, true, HttpStatusCode.Forbidden)]
+        [InlineData("admin/host/extensionBundle/v1/templates", true, true, false, true, false, HttpStatusCode.Unauthorized)]
+        [InlineData("admin/host/extensionBundle/v1/templates", true, true, true, false, true, HttpStatusCode.NotFound)]
+        [InlineData("admin/host/extensionBundle/v1/templates", true, false, false, false, true, HttpStatusCode.NotFound)]
+        [InlineData("admin/host/extensionBundle/v1/templates", true, true, false, true, true, HttpStatusCode.Forbidden)]
+        [InlineData("admin/vfs/host.json", true, true, true, false, true, HttpStatusCode.OK)]
+        [InlineData("admin/vfs/host.json", true, true, false, false, true, HttpStatusCode.Unauthorized)]
+        [InlineData("admin/vfs/host.json", true, true, true, false, false, HttpStatusCode.Unauthorized)]
+        public async Task AdminIsolation_ReturnsExpectedStatus(string uri, bool isAppService, bool enableIsolation, bool isPlatformInternal, bool bypassFE, bool addAuthKey, HttpStatusCode expectedStatus)
+        {
+            var environment = this._fixture.Host.WebHostServices.GetService<IEnvironment>();
+            string websiteInstanceId = environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId);
+
+            try
+            {
+                if (enableIsolation)
+                {
+                    environment.SetEnvironmentVariable(EnvironmentSettingNames.FunctionsAdminIsolationEnabled, "1");
+                    Assert.True(environment.IsAdminIsolationEnabled());
+                }
+
+                if (!isAppService)
+                {
+                    environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId, null);
+                    Assert.False(environment.IsAppService());
+                }
+                else
+                {
+                    Assert.True(environment.IsAppService());
+                }
+
+                HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, uri);
+
+                if (addAuthKey)
+                {
+                    await this._fixture.AddMasterKey(request);
+                }
+
+                if (isPlatformInternal)
+                {
+                    request.Headers.Add(ScriptConstants.AntaresPlatformInternal, "True");
+                }
+
+                if (!bypassFE)
```

Comment:
nit - maybe name "bypassFrontEnd" (if FE means front end :) ) for easier readability?